opentelemetry/src/sdk/trace/span_processor.rs

```diff
@@ -357,15 +419,45 @@ impl BatchSpanProcessor {
             config.max_export_batch_size = max_export_batch_size;
         }
 
+        let max_export_time_out = std::env::var(OTEL_BSP_EXPORT_TIMEOUT_MILLIS)
+            .map(|timeout| {
+                u64::from_str(&timeout).unwrap_or(OTEL_BSP_EXPORT_TIMEOUT_MILLIS_DEFAULT)
+            })
+            .unwrap_or(OTEL_BSP_EXPORT_TIMEOUT_MILLIS_DEFAULT);
+        config.max_export_timeout = time::Duration::from_millis(max_export_time_out);
+
         BatchSpanProcessorBuilder {
             config,
             exporter,
             spawn,
+            delay,
             interval,
         }
     }
 }
 
+async fn export_with_timeout<D, DS, E>(
+    time_out: time::Duration,
+    exporter: &mut E,
+    delay: &D,
+    batch: Vec<SpanData>,
+) -> ExportResult
+where
+    D: (Fn(time::Duration) -> DS) + Send + Sync + 'static,
+    DS: Future<Output = ()> + 'static + Send + Sync,
+    E: SpanExporter + ?Sized,
+{
+    let export = exporter.export(batch);
+    let timeout = delay(time_out);
+    pin_mut!(export);
+    pin_mut!(timeout);
+    // TODO: Surface error through global error handler
```

Comment:
I think the comment may be in the wrong place. This function returns the result. I think the reporting is necessary in places where we don't do anything with the result, e.g. in the Flush/Shutdown events of the batch span processor.